### PR TITLE
LF-3599: Review finances dashboard and check that new custom expenses and revenues function properly in these views

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

- update `saleController` to update `value` and `note` for general type and handle `crop_variety_sale` only when the type is crop sale (It will be properly done in [LF-3595](https://lite-farm.atlassian.net/browse/LF-3595))
- create `revenueFormTypes` constant and `getRevenueFormType` function
- update `GeneralRevenueForm` to be able to be used for edit view (fix default values, add delete button)
  Screenshot 2023-09-18 at 3 43 07 PM
- update `EditSale` to show `GeneralRevenueForm` based on `revenueType`
- update `ActualCropRevenue` (renamed to `ActualRevenueItem`) to show non crop revenue
   Screenshot 2023-09-18 at 3 37 03 PM
- update `getExpenseType` to show custom expense type name
   Screenshot 2023-09-19 at 2 24 57 PM
- skip a test for patchSales that cannot pass until the controller is updated

Other tweaks:
- correct persistedPaths for the revenue flow (remove `/manage_custom_revenues`)
- update the page to go on clicking "+ Add revenue"

Jira link: https://lite-farm.atlassian.net/browse/LF-3599

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

[LF-3595]: https://lite-farm.atlassian.net/browse/LF-3595?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/webapp/src/containers/Finances/ActualRevenueItem/index.jsx
Comment (SayakaOno) [inline packages/webapp/src/containers/Finances/ActualRevenueItem/index.jsx:1]: Please find the actual changes [here](https://github.com/LiteFarmOrg/LiteFarm/commit/e517b6aa5d2e83cad4f47d06e10d48df1be2e9c7).

Comment (Duncan-Brain) [reply packages/webapp/src/containers/Finances/ActualRevenueItem/index.jsx:1]: would "[git mv](https://git-scm.com/docs/git-mv)" have done it?

**Review by Duncan-Brain — APPROVED** [review]
Everything I can see looks good!

Sorry I have been slow to work on [LF-3628](https://lite-farm.atlassian.net/browse/LF-3628) so it is hard to test adding sales which is my fault.

**Review by kathyavini — APPROVED** [review]
This looks great!! I hope most can be reused smoothly for the new table view!!! 🤞

Thank you once again for the LF-3595 notes, I think it is pretty smoothly implementable now! The revenueFormType additions are so readable 👌

File: packages/webapp/src/containers/Finances/ActualRevenueItem/index.jsx
Comment (SayakaOno) [reply packages/webapp/src/containers/Finances/ActualRevenueItem/index.jsx:1]: I will try it next time, thank you for the link!!

**Pull request merged**